app/src/main/java/com/navapp/navigation/route/RouteHandler.java

```diff
@@ -0,0 +1,45 @@
+package com.navapp.navigation.route;
+
+import com.navapp.navigation.destination.Destination;
+import com.navapp.navigation.destination.data.Address;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+
+public class RouteHandler {
```

Comment:
What does `Handler` mean exactly?

Reply:
It interacts and does things with it?

Reply:
Both "does things with it" and `Handler` say exactly nothing about what happens the class. Be more specific.

Reply:
It's hard to explain, but basically, there are currently 3 classes related to routes:
1. Route- it's a data class, it contains all the stuff that a route has.

Reply:
What is "all functionality"?! This is still breaking the Single Responsibility Principal. The class should have a strongly-defined functionality.

Reply:
ffs it's navigation,you can also suggest what to do instead of bugging me over this.